### PR TITLE
add image load benchmark

### PR DESCRIPTION
**Pull request by jonahwilliams**
Adds a benchmark that can be used to show improvements for https://github.com/flutter/engine/pull/32999

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I need to run update-packages --force-upgrade but holding off

File: dev/benchmarks/microbenchmarks/lib/ui/image_bench.dart [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  assert(false, "Don't run benchmarks in debug mode! Use 'flutter run --release'.");
+
+  final Stopwatch watch = Stopwatch();
+  await benchmarkWidgets((WidgetTester tester) async {
+    watch.start();
+    for (int i = 0; i < 10; i += 1) {
+      await Future.wait(<Future<ui.ImmutableBuffer>>[
+        for (String asset in assets)
+          rootBundle.load(asset).then((ByteData data) {
+            return ui.ImmutableBuffer.fromUint8List(data.buffer.asUint8List());
```
Comment (dnfield) [inline dev/benchmarks/microbenchmarks/lib/ui/image_bench.dart:133]: My assumption is once the ImmutableBuffer.fromAsset change lands that'll be added here right?

I don't know if it makes sense to do here or to do in a separate benchmark, but WDYT of having something measuring the time it takes to `instantiateImageCodec` vs the new API?

Comment (jonahwilliams) [reply dev/benchmarks/microbenchmarks/lib/ui/image_bench.dart:133]: Yeah, i would just replace this code and we'd see the skia perf benchmark numbers change all at once.

> I don't know if it makes sense to do here or to do in a separate benchmark, but WDYT of having something measuring the time it takes to instantiateImageCodec vs the new API?

Assuming you already have the bytes loaded?

Comment (dnfield) [reply dev/benchmarks/microbenchmarks/lib/ui/image_bench.dart:93]: I'm trying to suggest a measurement for getting the immutable buffer, and a measurement for getting an image codec from the buffer.

Your change will eliminate copies on both ends, right?

Comment (jonahwilliams) [reply dev/benchmarks/microbenchmarks/lib/ui/image_bench.dart:93]: Well, on one end. I think calling ui.ImmutableBuffer.fromUint8List includes the copy though. The codec doesn't have any more copies that we can eliminate IIRC

Comment (dnfield) [reply dev/benchmarks/microbenchmarks/lib/ui/image_bench.dart:91]: ohhh I see now, you're already capturing that by calling `ImmutableBuffer.fromUint8List`. Ok.

**Review by dnfield — APPROVED** [review]

**Pull request merged**